public/app/plugins/panel/trend/TrendPanel.tsx

```diff
@@ -24,6 +32,22 @@ export const TrendPanel = ({
 }: PanelProps<Options>) => {
   const { sync } = usePanelContext();
 
+  const preparePlotFrameTimeless = (frames: DataFrame[], dimFields: XYFieldMatchers, timeRange?: TimeRange | null) => {
+    dimFields = {
+      ...dimFields,
+      x: (field, frame, frames) => {
+        // Need to fallback to first number field if no xField is set in options otherwise panel crashes 😬
+        const trendXFieldName = options.xField
+          ? options.xField
+          : frames[0].fields.find((field) => field.type === FieldType.number)?.name;
```

Comment:
this fallback should happen at the top of the panel if `options.xField` is not user-specified. then here we don't need to worry about doing the fallback on each `field` iteration of the x matcher.